src/Raven.Server/Indexing/LuceneVoronDirectory.cs

```diff
@@ -127,6 +126,12 @@ public override void DeleteFile(string name, IState s)
             if (readResult == null)
                 throw new FileNotFoundException("Could not find file", name);
 
+            if (_indexOutputFilesSummary.HasVoronWriteErrors)
```

Comment:
Move this to the top of the method.
Probably a good idea to do this on the Write as well

Reply:
Also, on Dispose, so it wouldn't try to create a new stream on another file?